### PR TITLE
Pricing table styling

### PR DESCRIPTION
**Pull request by software-project**
# Pull Request Template

## Issue

Addresses #1401 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds ResponsiveTable with base styling and ability to modify headers and cells.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

In Storybook. Please make sure both desktop and mobile views are working as expected.

- [x] Desktop View
- [x] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Review by seanmalbert — APPROVED** [review]
This looks good to me. @akegan , can you please make sure this meets your needs.

File: ui-components/src/tables/ResponsiveTable.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  income: "t.income",
+  rent: "t.rent",
+}
+
+const pricingRows = [
+  {
+    units: (
+      <>
+        <p className="font-semibold">Studio</p>
+        <p className="text-sm text-gray-700 hidden md:block">Waitlist</p>
```
Comment (akegan) [inline ui-components/src/tables/ResponsiveTable.stories.tsx]: I updated this in 9095599 to be consistent at mobile and desktop view

**Review by akegan — APPROVED** [review]
@seanmalbert, thanks for reviewing! It LGTM too. I just pushed a commit to change the availability content so that it's consistent for mobile and desktop. 

Hopefully that will fix the failed deploy to get all the tests passing too.

**Comment by akegan** [comment]
@seanmalbert it look like all of the netlify tests failed because the deploy failed? Do you know how to fix that issue?

**Comment by seanmalbert** [comment]
> @seanmalbert it look like all of the netlify tests failed because the deploy failed? Do you know how to fix that issue?
> 
> Edit: Nevermind, I see that there's a compilation error, I'll see if I can fix it:
> 

@akegan , let me know if you want me to take a look.

**Pull request merged**